import_bsp/BspGeneric.py

```diff
@@ -36,11 +41,27 @@ def pack_lightmaps(bsp, import_settings):
     color_scale = 1.0
     color_components = 4
     try:
-        path = bsp.bsp_path[:-len(".bsp")] + "\\"
-        lm_files = os.listdir(path)
-        lm_list = [path + file_name
-                    for file_name in lm_files
-                    if file_name.lower().startswith('lm_') and file_name.lower().endswith('.tga')]
+        path = bsp.bsp_path[:-len(".bsp")]
+        lm_file_list = os.listdir(path)
+
+        lm_list = []
+        while True:
```

Comment:
Hm maybe better something like:
`for index in range(len(lm_file_list))`
I also suggest to initialize the lm_list with a list of Nones and adding the lightmaps to the correct slots. Like that a lightmap can be missing and the algorithm should still "work"

Reply:
is there an easy way to know the amount of lightmap in advance?

Reply:
I added the `range(len(lm_file_list))` even if it's not really useful as this loop is basically a kind of `do while` loop.

Reply:
The [Q3 BSP format](http://www.mralligator.com/q3/#Lightmaps) seems to not store the number of lightmaps, since when there is internal lump, the total of lightmaps seems to be:

> length / sizeof(lightmap) records in the lump, where length is the size of the lump itself.

Also, `len(lm_file_list)` can't be used as the amount of lightmap, since there may be the file system may provide the same lightmap in multiple format.

This lightmap loader is already over-engineered. I don't know of idtech3 engines that are case-insensitive on direct filesystem access (they are case insensitive for files stored in pk3).

If we want to improve things, it would have to be implemented in a VFS, not there.